### PR TITLE
DIS detailed with more information

### LINKED ISSUE #284: Not enough details about DIS (Device Integrity Service)
The DIS component needs better definition. It is unclear what the component for smartphones is, how to interact with it, what are its functionality etc...

Clarify that the Strongbox Keymaster serves as the HSM (Hardware Security Module) for Android, providing various functionalities including Key Attestation. Clarify that the Secure Enclave serves as the HSM for iPhone, with DeviceCheck acting as a series of services that provides the ability to verify the authenticity of the phone, its keys, and its apps.

### PR DESCRIPTION
**Pull request by cmarco0**
This PR aims to solve the issue #284

Added more information about the DIS (Device Integrity Service):
-What is for Android and iOS.
-Specified key attestation functionality for Android and iOS
-availability on smartphones

### CODE REVIEW

File: docs/en/wallet-attestation.rst [hunk 1]
```diff
@@ -103,8 +103,13 @@ Wallet Instance Initialization and Registration
...
-    This service, specifically developed by the manufacturer, is already integrated within the Android or iOS SDKs, so there is no need for a predefined endpoint to access it. Moreover, as it is specifically developed in the mobile architecture, it does not need to be registered as a Federation Entity, through the national accreditation systems.
+    The service also incorporates details in the signed object, such as the device type, model, app version, operating system version, bootloader status, and other relevant information to assess the device has not been compromised. For Android the DIS is represented by **StrongBox Keymaster** which is a physical HSM installed directly on the motherboard, it has various feature, the one we are interested to is named `Key Attestation`_, developer can leverage its functionality by the usage of `Play Integrity API`_. *Key attestation* aims to provide a way to strongly determine if a key pair is hardware-backed, what the properties of the key are, and what constraints are applied to its usage.
```
Comment (grausof) [inline docs/en/wallet-attestation.rst]:
```
For Android the DIS is represented by **StrongBox Keymaster** which is a physical HSM installed directly on the motherboard, it has various feature, the one we are interested to is named `Key Attestation`_,
```
This is wrong. `Key Attestation` is not part of `StrongBox Keymaster` and can also be used with a Trusted Execution Environment (TEE).  The combined use of Key Attestation and Play Integrity Check allows you to create a DIS on Android which therefore guarantees both that the backend hardware key and the app are genuine.

File: docs/en/wallet-attestation.rst [hunk 2]
```diff
@@ -103,8 +103,13 @@ Wallet Instance Initialization and Registration
...
+    For Apple devices the DIS is represented by **Secure Enclave**, a dedicated secure subsystem integrated into Apple's SoCs. Apple iOS is more fragmented than Android, in this case exists a series of services named `DeviceCheck`_ which provide a framework and server interface to manage device-specific data securely, developer can leverage its functionality by the usage of the framework itself. *DeviceCheck* It can be used to attest to the integrity of the device, apps, and/or encryption keys generated on the device, ensuring they were created in a secure environment like Secure Enclave.
```
Comment (grausof) [inline docs/en/wallet-attestation.rst]: For Apple devices the DIS is represented only by `DeviceCheck`. Secure Enclave is just a system for storing keys securely on par with Strongbox or TEE. Furthermore, I don't agree with saying that Apple is more fragmented.

File: docs/en/wallet-attestation.rst (hunk 1 above)
Comment (cmarco0) [reply docs/en/wallet-attestation.rst]: Reading again the docs https://developer.android.com/privacy-and-security/keystore , key attestation is not part but it is only a supported feature by Stronghox Keymaster.

**Comment by peppelinux** [comment]
As discussed during the editor's call we agreed that this PR must be changed according to the revision made, aiming to hold only the additional clarifications that would aim to improve the specification

@cmarco0 ^

**Review by fventola-ipzs — APPROVED** [review]

**Review by grausof — APPROVED** [review]

File: docs/en/wallet-attestation.rst [hunk 3]
```diff
@@ -103,8 +103,10 @@ Wallet Instance Initialization and Registration
...
+    This services, specifically developed by the manufacturer, are already integrated within the Android or iOS SDKs, so there is no need for a predefined endpoint to access it. Moreover, as it is specifically developed in the mobile architecture, it does not need to be registered as a Federation Entity, through the national accreditation systems.
+    For Apple devices *Secure Enclave* is available since the iPhone 5s (2013).
+    *Strongbox Keymaster* is different in that, because each smartphone manufacturer must decide whether to include it or not.
```
Comment (peppelinux) [inline docs/en/wallet-attestation.rst]:
```suggestion
    For Android devices, the inclusion of **Strongbox Keymaster** may vary by each smartphone manufacturer, who decides whether to include it or not.
```

File: docs/en/wallet-attestation.rst [hunk 4]
```diff
@@ -103,8 +103,10 @@ Wallet Instance Initialization and Registration
...
+    This services, specifically developed by the manufacturer, are already integrated within the Android or iOS SDKs, so there is no need for a predefined endpoint to access it. Moreover, as it is specifically developed in the mobile architecture, it does not need to be registered as a Federation Entity, through the national accreditation systems.
```
Comment (peppelinux) [inline docs/en/wallet-attestation.rst]:
```suggestion
    These services, specifically developed by the manufacturer, are integrated within the Android or iOS SDKs, eliminating the need for a predefined endpoint to access them. Additionally, as they are specifically developed for mobile architecture, they do not need to be registered as Federation Entities through national accreditation systems.
```

**Pull request merged**